pages/blog/subscriptions-and-live-queries-real-time-with-graphql.mdx

```diff
@@ -0,0 +1,1150 @@
+import { createApplyLiveQueryPatch } from '@n1ru4l/graphql-live-query-patch';
+import { createClient } from 'graphql-ws/lib/use/ws';
+
+const client = createClient({
+  url: 'ws://localhost:3000/graphql',
+});
+
+const applyLiveQueryPatch = createApplyLiveQueryPatch()
+
+const asyncIterator = pipe(
+  makeAsyncIterableIteratorFromSink((sink) => {
+    const dispose = client.subscribe(
+      {
+        query: 'query @live { hello }',
+      },
+      {
+        next: sink.next,
+        error: sink.error,
+        complete: sink.complete,
+      }
+    );
+    return () => dispose();
+  }),
+  applyLiveQueryPatch
+);
+
+// stop live query after 5 seconds.
+setTimeout(() => {
+  asyncIterator.return();
+}, 5000);
+
+for await (const executionResult of asyncIterator) {
+  console.log(executionResult);
+}
+```
+
+So let's take a look at this implementation regarding the aspects we used before.
+
+1. Automatically update the clients. **Yes.** ✅
+
+The approach of pushing the invalidation responsibility to the server might at first seem like a drawback, but a smart abstraction (such as a ORM middleware), can result in pretty responsive applications.
+
+2. Efficiently updates clients. **Yes.** ✅
+
+In case our execution result payloads are getting too big we can easily enable JSON patches by adding a middleware.
+Furthermore, the middleware is totally live query implementation independent!
+That means if our projects requires a different implementation of the live query engine, the middleware can still be applied as long as the execution result is compatible with the live query execution result format.
+
+3. Flexibility. **High.** ✅
+
+Any database or third-party API can be used as long as we can somehow invalidate the resource (via some PubSub system etc.).
+As all incorporated libraries are pretty pluggable only a few can be used and others be replaced by something that might make more sense for a specific project.
+
+4. Adoptability. **High.** ✅
+
+This library can be added to any existing GraphQL.js schema without any hassle. Resource invalidation code can be added to the mutation resolvers over time gradually to reflect the needs of the frontend.
+The possibilities of resource invalidation are endless and the logic for those can be added incrementally.
+E.g. via a ORM middleware, in our mutation code or maybe even on our GraphQL gateway.
+
+[GraphQL Live Query libraries and example todo app that sync across all connected clients.](https://github.com/n1ru4l/graphql-live-query)
+
+[More information on how the `InMemoryLiveQueryStore` keeps track of resources here.](https://dev.to/n1ru4l/collecting-graphql-live-query-resource-identifier-with-graphql-tools-5fm5)
+
+## What's next?
+
+I hope that more people will try exploring the possibilities of GraphQL live queries. Eventually, we could even come up with an official RFC for the spec!
+
+There is a lot of more to explore such as
```

Comment:
I'd rather state: `There is a lot more to explore` - without the `of`.

Reply:
```suggestion
There is a lot more to explore such as
```

Reply:
thank you for all the typo fixes 😄